pkg/component/etcd/statefulset/values_helper.go

```diff
@@ -0,0 +1,696 @@
+
+		if val.OwnerCheck.Interval != nil {
+			command = append(command, "--owner-check-interval="+val.OwnerCheck.Interval.Duration.String())
+		}
+		if val.OwnerCheck.Timeout != nil {
+			command = append(command, "--owner-check-timeout="+val.OwnerCheck.Timeout.Duration.String())
+		}
+		if val.OwnerCheck.DNSCacheTTL != nil {
+			command = append(command, "--owner-check-dns-cache-ttl="+val.OwnerCheck.DNSCacheTTL.Duration.String())
+		}
+	}
+
+	if val.AutoCompactionMode != nil {
+		command = append(command, "--auto-compaction-mode="+string(*val.AutoCompactionMode))
+	} else {
+		command = append(command, "--auto-compaction-mode="+defaultAutoCompactionMode)
+	}
+
+	if val.AutoCompactionRetention != nil {
+		command = append(command, "--auto-compaction-retention="+string(*val.AutoCompactionRetention))
+	} else {
+		command = append(command, "--auto-compaction-retention="+defaultAutoCompactionRetention)
+	}
+
+	if val.EtcdSnapshotTimeout != nil {
+		command = append(command, "--etcd-snapshot-timeout="+val.EtcdSnapshotTimeout.Duration.String())
+	} else {
+		command = append(command, "--etcd-snapshot-timeout="+defaultEtcdSnapshotTimeout)
+	}
+
+	if val.EtcdDefragTimeout != nil {
+		command = append(command, "--etcd-defrag-timeout="+val.EtcdDefragTimeout.Duration.String())
+	} else {
+		command = append(command, "--etcd-defrag-timeout="+defaultEtcdDefragTimeout)
+	}
+
+	command = append(command, "--snapstore-temp-directory=/var/etcd/data/temp")
+	command = append(command, "--enable-member-lease-renewal=true")
+	command = append(command, "--etcd-process-name=etcd")
+
+	if heartBeatDuration := val.HeartbeatDuration; heartBeatDuration != nil {
+		command = append(command, "--k8s-heartbeat-duration="+heartBeatDuration.Duration.String())
+	} else {
+		command = append(command, "--k8s-heartbeat-duration="+defaultheartbeatDuration)
+	}
+
+	return command
+}
+
+func getEtcdEnvVar(val Values) []corev1.EnvVar {
+	protocol := "http"
+
+	if val.BackupTLS != nil {
+		protocol = "https"
+	}
+
+	endpoint := fmt.Sprintf("%s://%s-local:%d", protocol, val.EtcdName, pointer.Int32Deref(val.BackupPort, defaultBackupPort))
+
+	var env []corev1.EnvVar
+	env = append(env, getEnvVarFromValues("ENABLE_TLS", strconv.FormatBool(val.BackupTLS != nil)))
+	env = append(env, getEnvVarFromValues("BACKUP_ENDPOINT", endpoint))
+
+	return env
+}
+
+func getStsEnvVar(val Values) []corev1.EnvVar {
+	var env []corev1.EnvVar
+	env = append(env, getEnvVarFromFields("POD_NAME", "metadata.name"))
+	env = append(env, getEnvVarFromFields("POD_NAMESPACE", "metadata.namespace"))
+
+	if val.BackupStore == nil {
+		env = append(env, getEnvVarFromValues("STORAGE_CONTAINER", ""))
+		return env
+	}
+
+	storeValues := val.BackupStore
+
+	env = append(env, getEnvVarFromValues("STORAGE_CONTAINER", *storeValues.Container))
+
+	provider, err := utils.StorageProviderFromInfraProvider(val.BackupStore.Provider)
+	if err != nil {
+		return env
+	}
+
+	if provider == "S3" {
```

Comment:
It's not necessary to go through all cases as there can only be one provider. I suggest to have a `switch case` here.